internal/integration/api/etcd.go

```diff
@@ -0,0 +1,137 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// +build integration_api
+
+package api
+
+import (
+	"context"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/talos-systems/talos/internal/integration/base"
+	machineapi "github.com/talos-systems/talos/pkg/machinery/api/machine"
+	"github.com/talos-systems/talos/pkg/machinery/client"
+	"github.com/talos-systems/talos/pkg/machinery/config/types/v1alpha1/machine"
+	"github.com/talos-systems/talos/pkg/machinery/constants"
+)
+
+type EtcdSuite struct {
+	base.APISuite
+
+	ctx       context.Context
+	ctxCancel context.CancelFunc
+}
+
+// SuiteName ...
+func (suite *EtcdSuite) SuiteName() string {
+	return "api.EtcdSuite"
+}
+
+// SetupTest ...
+func (suite *EtcdSuite) SetupTest() {
+	if testing.Short() {
+		suite.T().Skip("skipping in short mode")
+	}
+
+	// make sure we abort at some point in time, but give enough room for Etcds
+	suite.ctx, suite.ctxCancel = context.WithTimeout(context.Background(), 30*time.Minute)
+}
+
+// TearDownTest ...
+func (suite *EtcdSuite) TearDownTest() {
+	if suite.ctxCancel != nil {
+		suite.ctxCancel()
+	}
+}
+
+// TestEtcdMoveLeader tests moving etcd leadership to another member.
+func (suite *EtcdSuite) TestEtcdMoveLeader() {
+	if suite.Cluster == nil {
+		suite.T().Skip("without full cluster state etcd test is not reliable (can't wait for cluster readiness in between resets)")
+	}
+
+	nodes := suite.DiscoverNodes().NodesByType(machine.TypeControlPlane)
+
+	var leader string
+
+	for _, node := range nodes {
+		resp, err := suite.Client.MachineClient.EtcdMoveLeader(client.WithNodes(context.TODO(), node), &machineapi.EtcdMoveLeaderRequest{})
+		suite.Require().NoError(err)
+
+		if resp.Messages[0].GetMember() != "" {
```

Comment:
probably you might need to assert here that `resp.Messages[0].Metadata.Error` is empty, as any errors are ignored now